hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsDistributor.java

```diff
@@ -97,12 +97,16 @@ public Map<AccountID, Long> payRewardsIfPending(
                     } while (beneficiary.deleted());
                 }
             }
-
-            if (!beneficiary.declineReward() && reward > 0) {
-                // even if reward is zero it will be added to rewardsPaid
-                applyReward(reward, beneficiary, writableStore);
-                rewardsPaid.merge(receiverId, reward, Long::sum);
+            // Even if the account has declineReward set or if reward is 0, it should still
+            // be added to the rewardsPaid map
```

Comment:
Good comment; could be improved by adding one more bit of context,
```
            // Even if the account has declineReward set or if reward is 0, it should still
            // be added to the rewardsPaid map because ONLY this map's keys have their
            // stakeAtStartOfLastRewardedPeriod value updated; and for some reward=0
            // situations it is critical to update stakeAtStartOfLastRewardedPeriod
```

Reply:
Changed. Thank you very much! 💯